http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -1145,10 +1145,22 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
   ): F[Vector[ListAppResponse]] = for {
     _ <- as.ask
     samResources = allClusters.flatMap(_.nodepools.flatMap(_.apps.map(_.samResourceId)))
-    samVisibleAppsOpt <- NonEmptyList.fromList(samResources).traverse { apps =>
-      authProvider.filterUserVisible(apps, userInfo)
+    // Partition apps by shared vs. private access scope and make a separate Sam call for each, then re-combine them.
+    // The reason we do this shared vs. private app access scope is represented by different Sam resource types/actions,
+    // and therefore needs different decoders to process the result list..
+    partition = samResources.partition(_.accessScope.exists(_ == AppAccessScope.WorkspaceShared))
+    samVisibleSharedAppsOpt <- NonEmptyList.fromList(partition._1).traverse { apps =>
+      authProvider.filterUserVisible(apps, userInfo)(implicitly, sharedAppSamIdDecoder, implicitly)
```

Comment:
What does the `(implicitly, sharedAppSamIdDecoder, implicitly)` call do? Just curious

Reply:
So `filterUserVisible` takes 3 implicit arguments:
```
(implicit
    sr: SamResource[R],
    decoder: Decoder[R],
    ev: Ask[F, TraceId]
  )
```
I want to _explicitly_ pass the `decoder` parameter. Scala doesn't let you pass implicits by name though -- so I can't do this:
```
 authProvider.filterUserVisible(apps, userInfo)(decoder = sharedAppSamIdDecoder)
```
`implicitly` is just a keyword to look up the value from the implicit context. I could have also written it this way I guess:
```
    samVisibleSharedAppsOpt <- NonEmptyList.fromList(partition._1).traverse { apps =>
      implicit val d = sharedAppSamIdDecoder
      authProvider.filterUserVisible(apps, userInfo)
    }
```